### DISCUSSION TITLE
Duplicate comments / stops on comments

### DISCUSSION THREAD

**Discussion by 403-html** [community/community#7766] [Copilot Conversations]
I have noticed that Copilot often stops on his comments and duplicates them. I know it is based on OpenAI, is it possible to increase the "Frequency penalty" of the model a bit? So that it doesn't repeat itself?

Other than that, it generally does a pretty good job of writing monotonic tests (with slight variations), which will probably be a point in favor of buying access later when it comes out officially.

**Comment by MulleDK19** [comment]
I doubt that would be a good idea. Code is repetitive. You don't want it to try to do things like start using double or float when it should be using int, just because it's used int previously.

**Reply by 403-html** [reply]
true, true - but I mean about comments, which are repeating

I know in practice I take care of this by starting writing code by myself after comments and getting autocompletion on the function itself but still sometimes it's useful to use Copilot to autocomplete comments because usually, it's accurate

**Reply by MulleDK19** [reply]
Don't use it to write comments for you, start the comments and have it complete them.

**Reply by 403-html** [reply]
Understood, but still in blank lines sometimes it's repeating the comments from above and even in "top 10" best solutions.

**Reply by MulleDK19** [reply]
The only time I see it as a problem is when it tries to write a comment instead of code. Like if you've described something with a few lines of comment, and it then just keeps writing comments instead of code.

**Reply by 403-html** [reply]
yes, that's the point

**Reply by MulleDK19** [reply]
I see.
